xpano/algorithm/algorithm.cc

```diff
@@ -101,14 +101,14 @@ std::vector<Pano> FindPanos(const std::vector<Match>& matches) {
   return result;
 }
 
-std::optional<cv::Mat> Stitch(const std::vector<cv::Mat>& images) {
+std::pair<int, std::optional<cv::Mat>> Stitch(const std::vector<cv::Mat>& images) {
```

Comment:
I think we can return the cv::Stitcher::Status directly, no need to do static_cast then.

We can also drop the optional from the second return value, since there is the return code as well.

Reply:
I think returning only **cv::Stitcher::Status** would create a restructuring need for other functions as well because the value returned through this function is also used in **stitcher_pipeline.cc** at line **96**. Instead of that, I was thinking about a tuple or a vector.

Reply:
I like the std::pair, no need for a tuple/vector.
I'd just make it `std::pair<cv::Stitcher::Status, cv::Mat>` instead. You can change the calling site in stitcher_pipeline.cc, that is ok.

Using plain int for status codes is not good coding practice (you cannot see the meaning of the specific values in code), and we can use the one from OpenCV since it is available.

In a larger project / later we would probably create a custom enum in the xpano namespace and translate the one from OpenCV to that.